changelog/gc_in_finalizer.dd

```diff
@@ -0,0 +1,18 @@
+$(REF inFinalizer, core, GC) was added
+
+A new function exposing information about the state of the GC was added.
+It returns `true` if the current thread is executing destructors (finalizers)
+of objects allocated on the GC heap that are no longer being referenced.
+
+Destructors of objects allocated on the GC heap presently have several
+limitations or oddities:
+* As destructors of all unreachable objects are ran in indeterministic order,
+  there's no guarantee that object members are alive during the object's
+  finalization
+* Allocation during finalization is disallowed
+* In contrast to stack allocated objects, partially constructed objects (object
+  whose constructor threw an exception) may be finalized
+
+$(REF inFinalizer, core, GC) can be used to efficiently guard against
+programming errors such as the above, or to empirically verify that certain
+objects are not left to be destroyed by the GC.
```

Comment:
On that last point, it's a bit confusing. You can't verify that they aren't *left* to be destroyed by the GC (which happens when you stop referencing an object, and isn't detectable), you can only verify that they aren't destroyed by the GC.

Perhaps "empirically verify if an object is being destroyed by the GC when it shouldn't be."

Reply:
Yeah, probably I didn't use the best words here, but I still think this covers my intention.

Of course, you can't know with 100% certainty if an object will be finalized by the GC as that's indeterministic (!= nondeterministic) and that's why I qualified it that by saying "empirically".

But I think you can verify if an object was left to be destroyed by the GC - unless you manually called `runFinalizers` no one else will call the object destructor while having `inFinalizer` also be true, unless the object was no longer referenced and was being collected by the GC.